src/basic/fs-util.h

```diff
@@ -163,3 +163,11 @@ static inline int xopenat_lock(int dir_fd, const char *path, int open_flags, Loc
 int link_fd(int fd, int newdirfd, const char *newpath);
 
 int linkat_replace(int olddirfd, const char *oldpath, int newdirfd, const char *newpath);
+
+static inline int at_flags_mangle_follow_nofollow(int flags) {
```

Comment:
i would drop the "follow" here in the name, to make clear what the result is. and maybe call this "normalize" rather than mangle?

`at_flags_normalize_nofollow()`

And then maybe later we should also add `at_flags_normalize_follow()` which normalizes things to the other direction?